<?xml version="1.0"?>
<launch>
  <!-- send additional description parameters -->
  <include file="$(find cob_arm_navigation)/launch/setup/planning_environment.launch" />

  <node pkg="planning_environment" type="environment_server" output="screen" name="cob3_environment_server">      
    <!--remap from="collision_map" to="collision_map_occ" /-->
	<!--remap from="collision_map_update" to="collision_map_occ_update" /-->
    
    <param name="group" type="string" value="arm" />
    <param name="allow_valid_collisions" type="bool" value="false" />

    <!--param name="collision_map_safety_timeout" type="double" value="100000.0" /-->
    <param name="joint_states_safety_timeout" type="double" value="1.0" />

    <param name="bounding_planes" type="string" value="0 0 1 -0.01" />

    <!--param name="pointcloud_padd" type="double" value="0.00" /-->

    <!-- it's enough to find one single contact to say that an collision occured -->
    <!-- don't spend unecessary effort to compute more contact points -->
    <param name="contacts_to_compute_for_display" type="int" value="1" />

	<param name="use_collision_map" type="bool" value="false" />
	
	<!-- global_frame necessary for planning with static obstacles, otherwise known objects move with base_footprint -->
	<param name="global_frame" type="string" value="/map" />
	<!--param name="global_frame" type="string" value="/base_footprint" /-->
	<!-- tf_safety_timeout necessary when global_frame is set -> planning_monitor_.includePose_ = true -> isPoseUpdated() in isEnvironmentSafe() -> tf_safety_timeout would be 0.0 by default -> planning fails -->
    <param name="tf_safety_timeout" type="double" value="1.0" />

    <rosparam command="load" file="$(find cob_arm_navigation)/config/environment_server_padding.yaml" />		

  </node>

</launch>

